be/src/exec/vectorized/chunks_sorter_topn.cpp

```diff
@@ -460,4 +464,28 @@ Status ChunksSorterTopn::_hybrid_sort_first_time(RuntimeState* state, Permutatio
     return Status::OK();
 }
 
+void ChunksSorterTopn::_rank_pruning() {
+    if (_topn_type != TTopNType::RANK) {
+        return;
+    }
+    if (_merged_segment.order_by_columns.empty()) {
+        return;
+    }
+    if (_merged_segment.chunk->num_rows() <= _get_number_of_rows_to_sort()) {
+        return;
+    }
+    const auto peer_group_start = static_cast<int>(_get_number_of_rows_to_sort()) - 1;
+    DCHECK_GE(peer_group_start, 0);
+    Tie tie(_merged_segment.chunk->num_rows(), 1);
+    vectorized::build_tie_for_columns(_merged_segment.order_by_columns, &tie);
```

Comment:
Would it be costive to `build_tie_for_columns` just for pruning ?

Reply:
Emm, to avoid overhead virtual function call, and may be benefit in some extreme cases.

And maybe it is expensive in most normal cases